climada/hazard/tc_tracks_synth.py

```diff
@@ -166,6 +206,170 @@ def _one_rnd_walk(track, ens_size, ens_amp0, ens_amp, max_angle, rnd_vec):
     return ens_track
 
 
+def _random_uniform_ac(n_ts, autocorr, time_step_h):
+    """Generate a series of autocorrelated uniformly distributed random numbers.
+
+    This implements the algorithm described here to derive a uniformly distributed
+    series with specified autocorrelation (here at a lag of 1 hour):
+    https://stats.stackexchange.com/questions/48086/
+        algorithm-to-produce-autocorrelated-uniformly-distributed-number
+    Autocorrelation is specified at a lag of 1 hour. To get a time series at a
+    different temporal resolution (time_step_h), an hourly time series is generated
+    and resampled (using linear interpolation) to the target resolution.
+
+    Parameters:
+        n_ts (int): length of the series
+        autocorr: autocorrelation (between -1 and 1)
+
+    Returns:
+        numpy.ndarray of length n
+    """
+    # generate autocorrelated 1-hourly perturbations, so first create hourly
+    #   time series of perturbations
+    n_ts_hourly_exact = n_ts * time_step_h
+    n_ts_hourly = int(np.ceil(n_ts_hourly_exact))
+    x = np.random.normal(size=n_ts_hourly)
+    theta = np.arccos(autocorr)
+    for i in range(1, len(x)):
+        x[i] = _h_ac(x[i - 1], x[i], theta)
+    # scale x to have magnitude [0,1]
+    x = (x + np.sqrt(3)) / (2 * np.sqrt(3))
+    # resample at target time step
+    x_ts = np.interp(np.arange(start=0, stop=n_ts_hourly_exact, step=time_step_h),
+                     np.arange(n_ts_hourly), x)
+    return x_ts
+
+
+@jit
+def _h_ac(x, y, theta):
```

Comment:
Missing docstring.

Reply:
I suggested the change away from `ensemsble_size` because to me this is confusing wording. The method is calc_rnd_walks, so it computes random walks and not an ensemble. Also, I was always confused whether after applying the method, the set of track now contains `ens_size` number of tracks or `en_size + 1`. The answer is en_size + 1` (it contains the original tracks) which I find confusing. Why would the ensemble size be larger than `ens_size`? That is why I suggested a name which say explicitly how many tracks are generate by `calc_rnd_walks`. 

Any ideas how to resolve this? (it is not really critical, so it can also stay at ens_size).

Reply:
Also fixed in commit e837bc88958cd756867fef5a7ef8d79571c8981c
(@chahank I guess your comment here was a reply to something else so I didn't deal with it here, but likely above).